group-coordinator/src/test/java/org/apache/kafka/coordinator/group/consumer/CurrentAssignmentBuilderTest.java

```diff
@@ -0,0 +1,548 @@
+        assertEquals(11, updatedMember.nextMemberEpoch());
+        assertEquals(mkAssignment(
+            mkTopicAssignment(topicId1, 3),
+            mkTopicAssignment(topicId2, 6)
+        ), updatedMember.assignedPartitions());
+        assertEquals(mkAssignment(
+            mkTopicAssignment(topicId1, 1, 2),
+            mkTopicAssignment(topicId2, 4, 5)
+        ), updatedMember.partitionsPendingRevocation());
+        assertEquals(mkAssignment(
+            mkTopicAssignment(topicId1, 4, 5),
+            mkTopicAssignment(topicId2, 7, 8)
+        ), updatedMember.partitionsPendingAssignment());
+    }
+
+    @Test
+    public void testTransitionFromRevokeToAssigning() {
+        Uuid topicId1 = Uuid.randomUuid();
+        Uuid topicId2 = Uuid.randomUuid();
+
+        ConsumerGroupMember member = new ConsumerGroupMember.Builder("member")
+            .setMemberEpoch(10)
+            .setPreviousMemberEpoch(10)
+            .setNextMemberEpoch(11)
+            .setAssignedPartitions(mkAssignment(
+                mkTopicAssignment(topicId1, 3),
+                mkTopicAssignment(topicId2, 6)))
+            .setPartitionsPendingRevocation(mkAssignment(
+                mkTopicAssignment(topicId1, 1, 2),
+                mkTopicAssignment(topicId2, 4, 5)))
+            .setPartitionsPendingAssignment(mkAssignment(
+                mkTopicAssignment(topicId1, 4, 5),
+                mkTopicAssignment(topicId2, 7, 8)))
+            .build();
+
+        assertEquals(ConsumerGroupMember.MemberState.REVOKING, member.state());
+
+        Assignment targetAssignment = new Assignment(mkAssignment(
+            mkTopicAssignment(topicId1, 3, 4, 5),
+            mkTopicAssignment(topicId2, 6, 7, 8)
+        ));
+
+        ConsumerGroupMember updatedMember = new CurrentAssignmentBuilder(member)
+            .withTargetAssignment(11, targetAssignment)
+            .withCurrentPartitionEpoch((topicId, partitionId) -> 10)
+            .withOwnedTopicPartitions(requestFromAssignment(mkAssignment(
+                mkTopicAssignment(topicId1, 3),
+                mkTopicAssignment(topicId2, 6))))
+            .build();
+
+        assertEquals(ConsumerGroupMember.MemberState.ASSIGNING, updatedMember.state());
+        assertEquals(10, updatedMember.previousMemberEpoch());
+        assertEquals(11, updatedMember.memberEpoch());
+        assertEquals(11, updatedMember.nextMemberEpoch());
+        assertEquals(mkAssignment(
+            mkTopicAssignment(topicId1, 3),
+            mkTopicAssignment(topicId2, 6)
+        ), updatedMember.assignedPartitions());
+        assertEquals(Collections.emptyMap(), updatedMember.partitionsPendingRevocation());
+        assertEquals(mkAssignment(
+            mkTopicAssignment(topicId1, 4, 5),
+            mkTopicAssignment(topicId2, 7, 8)
+        ), updatedMember.partitionsPendingAssignment());
+    }
+
+    @Test
+    public void testTransitionFromRevokeToStable() {
```

Comment:
Do we have a test where the original state was to revoke some partitions and the target becomes those same revoked ones (so we should go to stable I think)?

Reply:
Nope. Let me add it.